packages/notifi-react-card/lib/components/WalletList/ConnectWalletRow.tsx

```diff
@@ -47,31 +63,85 @@ export const ConnectWalletRow: React.FC<ConnectWalletRowProps> = ({
     return addressEllipsis(walletParams.walletPublicKey);
   }, [walletParams]);
 
-  const connectWallet = useCallback(async () => {
-    await subscribeWallet({
-      walletParams,
-      connectWalletConflictResolutionTechnique: 'FAIL',
-    });
-  }, [subscribeWallet, walletParams]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isConnectedElsewhere, setIsConnectedElsewhere] = useState(false);
+
+  const connectWallet = useCallback(
+    async (technique: 'FAIL' | 'DISCONNECT_AND_CLOSE_OLD_ACCOUNT') => {
+      setIsLoading(true);
+      try {
+        await subscribeWallet({
+          walletParams,
+          connectWalletConflictResolutionTechnique: technique,
+        });
+        setIsConnectedElsewhere(false);
+      } catch (e: unknown) {
+        if (e instanceof GqlError) {
+          const alreadyConnectedError =
+            findError(
+              e.errors,
+              'ERROR_WALLETCONNECT_ALREADY_CONNECTED_TO_ANOTHER_ACCOUNT_AND_LAST',
+            ) ??
+            findError(
+              e.errors,
+              'ERROR_WALLETCONNECT_ALREADY_CONNECTED_TO_ANOTHER_ACCOUNT',
+            );
+          if (alreadyConnectedError !== undefined) {
+            setIsConnectedElsewhere(true);
+          }
+        }
+
+        throw e;
+      } finally {
+        setIsLoading(false);
+      }
+    },
+    [subscribeWallet, walletParams],
+  );
 
-  if (isConnected) {
-    return (
-      <div className="NotifiVerifyItem">
-        <p className="NotifiVerifyPublicKey">{shortenedAddress}</p>
-      </div>
-    );
-  }
   return (
-    <div className="NotifiVerifyItem">
-      <p className="NotifiVerifyPublicKey">{shortenedAddress}</p>
-      <button
-        className="NotifiVerifyButton"
-        onClick={() => {
-          connectWallet();
-        }}
-      >
-        Connect
-      </button>
+    <div className="ConnectWalletRow__container">
+      <div className="ConnectWalletRow__topRow">
+        <p className="ConnectWalletRow__publicKey">{shortenedAddress}</p>
+        {isConnected || isConnectedElsewhere ? null : (
+          <button
+            disabled={isLoading || isConnectedElsewhere}
+            className="ConnectWalletRow__button"
+            onClick={() => {
+              connectWallet('FAIL').catch((e) => {
+                console.log('Error connecting wallet', e);
+              });
+            }}
+          >
+            Verify
+          </button>
+        )}
+        {isConnected ? (
+          <p className="ConnectWalletRow__verified">Verified</p>
+        ) : null}
+      </div>
+      {isConnectedElsewhere ? (
```

Comment:
will they have the option to reconnect on notifi hub?

Reply:
No, just like when disconnecting from the hub, they'll lose access to the account.